src/test/java/org/prebid/server/bidder/tappx/TappxBidderTest.java

```diff
@@ -0,0 +1,228 @@
+
+        final BidRequest bidRequestEmptyEndpoint = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .ext(mapper.valueToTree(ExtPrebid.of(null,
+                                ExtImpTappx.of("host", "tappxkey", "")))).build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> emptyHostResult = TappxBidder.makeHttpRequests(bidRequestEmptyHost);
+        final Result<List<HttpRequest<BidRequest>>> emptyTappxKeyResult = TappxBidder.makeHttpRequests(bidRequestEmptyTappxKey);
+        final Result<List<HttpRequest<BidRequest>>> emptyEndpointResult = TappxBidder.makeHttpRequests(bidRequestEmptyEndpoint);
+
+        // then
+        assertThat(emptyHostResult.getErrors()).hasSize(1);
+        assertThat(emptyHostResult.getErrors().get(0).getMessage()).startsWith("Tappx host undefined");
+        assertThat(emptyTappxKeyResult.getErrors()).hasSize(1);
+        assertThat(emptyTappxKeyResult.getErrors().get(0).getMessage()).startsWith("Tappx tappxkey undefined");
+        assertThat(emptyEndpointResult.getErrors()).hasSize(1);
+        assertThat(emptyEndpointResult.getErrors().get(0).getMessage()).startsWith("Tappx endpoint undefined");
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorIfResponseBodyCouldNotBeParsed() {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+        // when
+        final Result<List<BidderBid>> result = TappxBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors().get(0).getMessage()).startsWith("Failed to decode: Unrecognized token");
+        assertThat(result.getErrors().get(0).getType()).isEqualTo(BidderError.Type.bad_server_response);
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseIsNull() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall(null,
+                mapper.writeValueAsString(null));
+
+        // when
+        final Result<List<BidderBid>> result = TappxBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfBidResponseSeatBidIsNull() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = givenHttpCall(null,
+                mapper.writeValueAsString(BidResponse.builder().build()));
+
+        // when
+        final Result<List<BidderBid>> result = TappxBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerBidIfBannerIsPresent() throws JsonProcessingException {
```

Comment:
As I've noted above, this is not the behavior they have for bid type selection. Please remake this test to the case `makeBidsShouldReturnDefaultBannerBid` (when imp has no video) and add additional case `makeBidsShouldReturnVideoBidIfVideoIsPresent` to reflect required logic.

Files and especially tests generated by bidder generator are good (although you still need to check them) for specific cases, which are pretty much tailor-made for bidder generator. As long as there are some changes that don't fit bidder generator's cases - always re-check the files and add specific test cases to cover what wasn't covered by generator.